components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/utils/RESTAPIAdminClient.java

```diff
@@ -102,21 +108,112 @@ public void updateApi(APITemplateBuilder builder, String tenantDomain) throws Ax
+					!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
 				restApiAdminStub.deleteApiForTenant(qualifiedName, tenantDomain);
 			} else {
 				restApiAdminStub.deleteApi(qualifiedName);
 			}
-			
+
 		} catch (Exception e) {
 			throw new AxisFault("Error while deleting WebApp", e);
 		}
 	}
+
+	/**
+	 * Adds non-versioned web app configuration to the gateway
+	 *
+	 * @param builder
+	 * @param tenantDomain
+	 * @throws AxisFault
+	 */
+	public void addNonVersionedWebApp(APITemplateBuilder builder, String tenantDomain)
+			throws AxisFault {
+
+		try {
+			String apiConfig = builder.getConfigStringForNonVersionedWebAppTemplate();
+			if (tenantDomain != null && !("").equals(tenantDomain)
+					&& !tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+				restApiAdminStub.addApiForTenant(apiConfig, tenantDomain);
+			} else {
+				restApiAdminStub.addApiFromString(apiConfig);
+			}
+		} catch (Exception e) {
+			throw new AxisFault("Error publishing non-versioned web app to the gateway", e);
+		}
+	}
+
+	/**
+	 * Updates non-versioned web app configuration in the gateway
+	 *
+	 * @param builder
+	 * @param tenantDomain
+	 * @throws AxisFault
+	 */
+	public void updateNonVersionedWebApp(APITemplateBuilder builder, String tenantDomain)
+			throws AxisFault {
+		try {
+			String apiConfig = builder.getConfigStringForNonVersionedWebAppTemplate();
+			if (tenantDomain != null && !("").equals(tenantDomain) &&
+					!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+
+				restApiAdminStub.updateApiForTenant(qualifiedNonVersionedWebAppName, apiConfig,
+													tenantDomain);
+			} else {
+				restApiAdminStub.updateApiFromString(qualifiedNonVersionedWebAppName, apiConfig);
+			}
+		} catch (Exception e) {
+			throw new AxisFault("Error while updating non-versioned web app in the gateway", e);
+		}
+	}
+
+	/**
+	 * Deletes non-versioned web app configuration form the gateway
+	 *
+	 * @param tenantDomain
+	 * @throws AxisFault
+	 */
+	public void deleteNonVersionedWebApp(String tenantDomain) throws AxisFault {
+		try {
+			if (tenantDomain != null && !("").equals(tenantDomain) &&
+					!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
+				restApiAdminStub.deleteApiForTenant(qualifiedNonVersionedWebAppName, tenantDomain);
+			} else {
+				restApiAdminStub.deleteApi(qualifiedNonVersionedWebAppName);
+			}
+		} catch (Exception e) {
+			throw new AxisFault("Error while deleting non-versioned web app from the gateway", e);
+		}
+	}
+
+	/**
+	 * Returns the non-versioned web app configuration from the gateway
+	 *
+	 * @param tenantDomain
+	 * @return
+	 * @throws AxisFault
+	 */
+	public APIData getNonVersionedWebAppData(String tenantDomain) throws AxisFault {
+		try {
+			APIData apiData;
+			if (tenantDomain != null && !("").equals(tenantDomain)
+					&& !tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
```

Comment:
here null & empty string check is redundant because, when `tenantDomain` equals to `MultitenantConstants.SUPER_TENANT_DOMAIN_NAME` it cannot be null or empty.
see commit 50d5da1ffa1fffacf500b8005728d72302726392 of PR #153